plugins/Installation/HostPortExtractor.php

```diff
@@ -0,0 +1,182 @@
+             */
+            if (
+                count($listOfTextInsideSquareBrackets) > 0 &&
+                filter_var(
+                    $listOfTextInsideSquareBrackets[0],
+                    FILTER_VALIDATE_IP,
+                    FILTER_FLAG_IPV6
+                ) !== false
+            ) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * Extracts the Host and Port from a user provided database server host,
+     * assuming the provided host is a IPv6 address, with or without a port
+     *
+     * @param string $dbHost The user provided database server host
+     * @throws InvalidArgumentException if the provided dbHost is not a valid IPv6
+     * @return HostPortExtractor The extracted Host & Port
+     */
+    private static function extractIPv6(string $dbHost): HostPortExtractor
+    {
+        if (!self::isIPv6($dbHost)) {
+            throw new InvalidArgumentException('$dbHost must be a valid IPv6 address');
+        }
+
+        // find and extract the text inside square brackets: []
+        preg_match_all('/\[(.*?)\]/', $dbHost, $matches);
+        $listOfTextInsideSquareBrackets = $matches[1];
+
+        $port = '';
+
+        // extract text after closing bracket to search for port
+        $components = explode(']', $dbHost);
+        if (count($components) > 1) {
+            // check if the text is a valid port e.g. ':3000'
+            preg_match('/^(:\d+)$/', $components[1], $portMatches);
+            if (count($portMatches) > 0) {
+                // only save the port number, not the colon
+                $port = substr($portMatches[1], 1);
+            }
+        }
+        // db connector requires IPv6 to be encased in square brackets
+        $host = '[' . $listOfTextInsideSquareBrackets[0] . ']';
+
+        return new HostPortExtractor($host, $port);
+    }
+
+    /**
+     * Determines if the provided host is a Unix Socket
+     *
+     * @param string $dbHost The user provided database server host
+     * @return boolean Whether the provided host is a unix socket
+     */
+    private static function isUnixSocket(string $dbHost): bool
+    {
+        return (strpos($dbHost, '/') !== false);
+    }
+
+    /**
+     * Extracts the host & port from the user provided database server host,
+     * assuming the provided host is a unix socket
+     *
+     * @param string $dbHost The user provided database server host
+     * @throws InvalidArgumentException if the provided dbHost is not a valid Unix Socket
+     * @return HostPortExtractor The extracted Host & Port
+     */
+    private static function extractUnixSocket(string $dbHost): HostPortExtractor
+    {
+        if (!self::isUnixSocket($dbHost)) {
+            throw new InvalidArgumentException('$dbHost must be a valid Unix Socket');
+        }
+
+        $portIndex = strpos($dbHost, '/');
+            // unix_socket=/path/sock.n
+        $port = substr($dbHost, $portIndex);
```

Comment:
Not 100% sure here, are we considering the socket being the port here? Should the variables be named differently or perhaps adjust the comment explaining the specificity of this?

Reply:
The DB connector requires unix sockets to be in the 'port' field. I will add a comment explaining this.

Reply:
I have made this change.